rust/xaynet-analytics/src/database/screen_route/data_model.rs

```diff
@@ -0,0 +1,37 @@
+use chrono::{DateTime, Utc};
+use isar_core::object::{data_type::DataType, object_builder::ObjectBuilder};
+use std::vec::IntoIter;
+
+use crate::database::common::{FieldProperty, IsarAdapter};
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct ScreenRoute {
+    pub name: String,
+    pub created_at: DateTime<Utc>,
+    pub object_id: Option<String>,
```

Comment:
is it because we want control over it, or is it because then `ScreenRouteRepo::add` needs to mutate the object?

when we `put` we can pass an Option<ObjectId> (right now we don't), but it has to be an `ObjectId`, not `String`. `ObjectId`s can be generated with:

`pub fn new(prefix: u16, time: u32, counter: u32, rand: u32) -> Self { ... }` 

where `time` and `counter` are big endians and prefix must not be the same as the `IsarCollection.id`, which we can't set but we have read access.

i could add it to `IsarDb`, so limit how much we mix our stuff with `Isar`'s, but `ScreenRouteRepo::add` will still need a mutable object. 

Does it make sense?

Reply:
I don't like that `add` takes a `&mut`. But also the semantic is not clear. Having an `AnalyticsEvent { screen_route: Some(ScreenRoute { object_id: None})}` is stored on db in the same way as an `AnalyticsEvent { screen_route: None}`.

If we have one single entry point to get `ScreenRoute`s from the app we can have there a struct without `object_id`, we can then insert them in the database and have a `ScreenRoute` with `object_id: String`.
There could be still the problem on how to recognize that two `ScreenRoute` from the app are the same and that it is already in the db.

Reply:
ok removed it after the upgrade and following the call we had